libethcore/BlockHeader.cpp

```diff
@@ -204,6 +204,7 @@ void BlockHeader::populateFromParent(BlockHeader const& _parent)
     m_parentHash = _parent.m_hash;
     m_gasLimit = _parent.m_gasLimit;
     m_difficulty = _parent.m_difficulty;
+    m_timestamp = _parent.m_timestamp + 1;
```

Comment:
Why is this needed?

Reply:
I think you should put `header.setTimestamp()` where you construct the block, instead of here.

This change could affect basically everything in the client. And looking at the code it seems that after this all our PoW mining will produce timestamps incremented from parent instead of real timestamps.

Also - better do this in a separate PR, this is not related to the test hashes.